diff --git a/test/rubygems/test_gem.rb b/test/rubygems/test_gem.rb
index dd2440b068..79ba2c401b 100644
--- a/test/rubygems/test_gem.rb
+++ b/test/rubygems/test_gem.rb
@@ -2139,6 +2139,8 @@ def with_local_bundler_at(path)
 
     # If bundler gemspec exists, pretend it's installed
     bundler_gemspec = File.expand_path("../../bundler/bundler.gemspec", __dir__)
+    bundler_gemspec = File.expand_path("../../lib/bundler/bundler.gemspec", __dir__) unless File.exist?(bundler_gemspec)
+    
     if File.exist?(bundler_gemspec)
       target_gemspec_location = "#{path}/specifications/bundler-#{Bundler::VERSION}.gemspec"
 
